plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -579,16 +585,87 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
     }
 
-    protected static String likeToRegexp(String like)
+    protected static boolean checkArgument(Call call, int index, Predicate<ConnectorExpression> predicate)
+    {
+        return predicate.test(call.getArguments().get(index));
+    }
+
+    protected static boolean isLikeCall(Call call)
+    {
+        if (!LIKE_PATTERN_FUNCTION_NAME.equals(call.getFunctionName())) {
+            return false;
+        }
+
+        if (!checkArgument(call, 0, Variable.class::isInstance)
+            && checkArgument(call, 1, Constant.class::isInstance)) {
+            return false;
```

Comment:
"isLikeCall" method shouldn't return false here. it **is a like call**, just one that is not supported.

while you can have `isSupportedLikeCall` method, it's better to have "is supported" conditions together with the code that consumes the information.

I'd recommend extracting whole `if (isLikeCall(call)) { ...` to a method.
Make this a prep commit.